src/google/protobuf/compiler/csharp/csharp_field_base.h

```diff
@@ -60,8 +60,8 @@ class FieldGeneratorBase : public SourceGeneratorBase {
   virtual void GenerateExtensionCode(io::Printer* printer);
   virtual void GenerateMembers(io::Printer* printer) = 0;
   virtual void GenerateMergingCode(io::Printer* printer) = 0;
-  virtual void GenerateParsingCode(io::Printer* printer) = 0;
-  virtual void GenerateSerializationCode(io::Printer* printer) = 0;
+  virtual void GenerateParsingCode(io::Printer* printer, bool use_buffer_serialization) = 0;
```

Comment:
Why not add a separate method like `GenerateBufferParsingCode` instead of adding a bool switch parameter?

Reply:
Yup I can give this a try.